Use a different approach to fix the smarty bug

mitya57 opened the pull request:
Just update `HTML_RE` so that it doesn't act on `<<...>>`.

mitya57 committed:
Revert "smarty: Use a separate processor for angled quotes"

That commit caused a regression where `<<` and `>>` inside code
blocks were wrongly replaced with double quotes.

This reverts commit 5029d829c1532f31adc9acbf54d88bb469a69a4f.

mitya57 committed:
Add a test to make sure quotes processor does not touch code blocks

waylan commented on markdown/inlinepatterns.py:
```diff
@@ -151,7 +151,7 @@ def build_inlinepatterns(md_instance, **kwargs):
 
 # <...>
-HTML_RE = r'(\<([a-zA-Z/][^\>]*?|\!--.*?--)\>)'
+HTML_RE = r'(\<([a-zA-Z/][^\>]*?|\!--.*?--)\>)(?!\>)'
```
I think this is a sensible approach when the smarty extension is in use, but the default behavior should not be changed when smarty is not enabled. I would expect that the smarty extension would replace the HTML pattern with a new one if angle quotes are enabled. Otherwise, leave it unchanged.

And we should have a test for each state (enabled and not enabled). The not enabled state should return the same results as we would get without smarty before this patch was applied.

mitya57 replied:
Is `<foo>>` really a valid HTML tag?

waylan replied:
No, of course not. But the parser is not concerned with whether raw HTML is "valid". Without smarty's angle quotes enabled, I expect the same default behavuor we have now: `<foo>>` becomes `<foo>&gt;`.

mitya57 committed:
Override regex for html inline pattern when smart_angled_quotes is true

Add a restriction that the closing angled quote should not be duplicate,
so that we can use our own pattern for handling duplicate quotes.

mitya57 committed:
Add a test for duplicate angled quotes without smarty extension

mitya57 commented:
@waylan I have now updated the pull request as you suggested.
